src/server/v2.0/handler/ldap.go

```diff
@@ -0,0 +1,96 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/controller/ldap"
+	"github.com/goharbor/harbor/src/lib/errors"
+	ldapModel "github.com/goharbor/harbor/src/pkg/ldap/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/ldap"
+)
+
+type ldapAPI struct {
+	BaseAPI
+	ctl ldap.Controller
+}
+
+func newLdapAPI() *ldapAPI {
+	return &ldapAPI{ctl: ldap.Ctl}
+}
+
+func (l *ldapAPI) PingLdap(ctx context.Context, params operation.PingLdapParams) middleware.Responder {
+	if err := l.RequireSystemAccess(ctx, rbac.ActionRead, rbac.ResourceConfiguration); err != nil {
```

Comment:
Will `rbac.ActionUpdate` on the `rbac.ResourceConfiguration` be better than `rbac.ActionRead` ?

The ping is helpful only when the user can update the configuration (LDAP settings).